### PR TITLE
Add support for specifying the anomaly ref year

### PR DESCRIPTION
**Pull request by xylar**
This is useful for runs that include a spin-up phase and anomalies are only of interest after the spin-up has completed (e.g. the last CORE-II cycle in a G-case E3SM run).

### CODE REVIEW

**Comment by xylar** [comment]
@vanroekel, do you feel comfortable testing and reviewing this PR?

**Comment by vanroekel** [comment]
@xylar sure I can test this.  But I can't do a thorough review until next week.

**Comment by xylar** [comment]
@vanroekel, that's fine.  I'm mostly just wanting to make sure this doesn't fall off the radar completely.  If you'd rather someone else review it, feel free to suggest someone.

**Comment by vanroekel** [comment]
@xylar I have tested this on anvil and am confused by the behavior.  I think the reference for anomalies is the average of timeSeriesStart to timeSeriesEnd for all combinations.  In the compute_moving_avg_anomaly_from_start function, if no endDate is provided to the dsStart open_mpas_dataset call, it appears to return the entire record.  Please correct me if I'm reading this wrong.

In my tests, I've used a time series window of years 10-15.  I tried many different anomalyRefYears (1-9) and the results were visually identical for OHC.  I also tried moving the years1-9 data to another folder and rerunning and no error was thrown.  I also tried having a refDate way past the end of the record (e.g. year 5000) and no error was thrown and results stayed consistent.

I think I have some code changes that address what I've noticed.  Would you like me to push them here?  Or somewhere else?

**Comment by xylar** [comment]
Sounds like my changes, made in a rush for the v1 paper, are buggy. Sorry. Please push your fix here and I'll take a look and do some proper testing this time.

File: mpas_analysis/ocean/compute_anomaly_subtask.py [hunk 1]
```diff
@@ -148,13 +148,20 @@ def run_task(self):  # {{{
         if config.has_option('timeSeries', 'anomalyRefYear'):
             anomalyYear = config.getint('timeSeries', 'anomalyRefYear')
             anomalyRefDate = '{:04d}-01-01_00:00:00'.format(anomalyYear)
+            anomalyEndDate = '{:04d}-12-31_23:59:59'.format(anomalyYear)
         else:
             anomalyRefDate = get_simulation_start_time(self.runStreams)
+            anomalyYear = int(anomalyRefDate[0:4])
+            anomalyEndDate = '{:04d}-12-31_23:59:59'.format(anomalyYear)
 
+        self.logger.info(anomalyRefDate)
```
Comment (xylar) [inline mpas_analysis/ocean/compute_anomaly_subtask.py]: Debug command? Can I remove this?

File: mpas_analysis/ocean/compute_anomaly_subtask.py [hunk 2]
```diff
@@ -148,13 +148,20 @@ def run_task(self):  # {{{
...
+
+        self.logger.info(self.inputFile)
```
Comment (xylar) [inline mpas_analysis/ocean/compute_anomaly_subtask.py]: Debug command? Can I remove this?

File: mpas_analysis/shared/time_series/mpas_time_series_task.py [hunk 3]
```diff
@@ -168,12 +168,14 @@ def setup_and_check(self):  # {{{
             anomalyStartDate = get_simulation_start_time(self.runStreams)
             anomalyYear = int(anomalyStartDate[0:4])
 
+        self.runMessage = '\n anomYear = {}'.format(anomalyYear)
```
Comment (xylar) [inline mpas_analysis/shared/time_series/mpas_time_series_task.py]: Debug command? Can I remove this?

File: mpas_analysis/shared/time_series/mpas_time_series_task.py [hunk 4]
```diff
@@ -168,12 +168,14 @@ def setup_and_check(self):  # {{{
...
         anomalyEndDate = '{:04d}-12-31_23:59:59'.format(anomalyYear)
         firstYearInputFiles = self.historyStreams.readpath(
                 streamName, startDate=anomalyStartDate,
                 endDate=anomalyEndDate,
                 calendar=self.calendar)
         for fileName in firstYearInputFiles:
+            self.runMessage = fileName
```
Comment (xylar) [inline mpas_analysis/shared/time_series/mpas_time_series_task.py]: Debug command? Can I remove this?

**Comment by xylar** [comment]
After @vanroekel's fix, I get plots like the following:
sanomalyz_global_gmpas-qu240
This is with `anomalyRefYear = 3`, so looks exactly as it should.

**Comment by xylar** [comment]
Thanks again @vanroekel.  I did a "fixup" of your commit where I removed some debugging commands.  My test on my laptop passed.  If you want to do any additional testing, feel free. Please mark this as "approved" when you think it's ready.

**Comment by xylar** [comment]
Thanks, @vanroekel.  Merging.

**Pull request merged**